### DISCUSSION TITLE
microstructure with different grains morphologies

### DISCUSSION THREAD

**Discussion by AGHz97** [neperfepx/neper#644] [Q&A] [answered]
Dear Romain, 
I would like to generate a microstructure with equiaxed grains in the first half (i.e. for z in [0, Lz/2]) and elongated grains along y in the second half (i.e. for z in [Lz/2, Lz]), while specifying the distribution law of diameters in each region (log-normal for example). Could you help me with the command to run? 
Thank you very much for your help.

**Comment by rquey** [comment, marked as answer]
I would proceed in several steps to generate the equiaxed grains on one hand and the elongated grains on the other hand, and then put them together.

Here is an example:

```
neper -T -n 500 -domain "cube(1,1,0.5)" -morpho gg -morphooptistop eps=1e-2 -statseed x,y,z,w -o part1
awk '{print $1,$2,$3}' part1.stseed > part1.coo
awk '{print $4}' part1.stseed > part1.w

neper -T -n 100 -dim 2 -morpho gg -morphooptistop eps=1e-2 -statseed x,y,z,w -o part2
awk '{print $1,$2,0.6}' part2.stseed > part2.coo
awk '{print $4}' part2.stseed > part2.w

cat part1.coo part2.coo > coo
cat part1.w part2.w > w

neper -T -n 600 -morphooptiini "coo:file(coo),weight:file(w)" -o full

neper -V full.tess -print img
```
![image](https://user-images.githubusercontent.com/2659376/220630537-94f3119f-f8f7-4852-ae2f-4478782448e5.png)

**Reply by MROHOMAN** [reply]
Hello @rquey ,
If I use neper -T -n 100 -dim 2 -morpho gg -morphooptistop eps=1e-2 -statseed x,y,z,w -o part2  it does not generate any columnar grains.

**Reply by rquey** [reply]
Run the full set of commands. `full.tess` is the final tessellation.
